Shared/mods/deathmatch/logic/physics/CLuaPhysicsRigidBody.cpp

```diff
@@ -0,0 +1,84 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/logic/lua/CLuaPhysicsRigidBody.cpp
+ *  PURPOSE:     Lua rigid body class
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include <StdInc.h>
+
+CLuaPhysicsRigidBody::CLuaPhysicsRigidBody(CLuaPhysicsShape* pPhysicsShape, float fMass, CVector vecLocalInertia, CVector vecCenterOfMass)
+    : CLuaPhysicsWorldElement(EIdClass::RIGID_BODY), m_pPhysicsShape(pPhysicsShape)
+{
+    m_pMotionState = std::make_unique<MotionState>();
+
+    m_pRigidBodyProxy = CPhysicsRigidBodyProxy::New(m_pPhysicsShape, fMass, vecLocalInertia, vecCenterOfMass, m_pMotionState.get());
+    m_pRigidBodyProxy->setUserPointer((void*)this);
+    m_pRigidBodyProxy->setUserIndex(EIdClass::RIGID_BODY);
+    pPhysicsShape->AddRigidBody(this);
+}
+
+void CLuaPhysicsRigidBody::SetPosition(CVector vecPosition)
+{
+    btTransform& transform = m_pRigidBodyProxy->getWorldTransform();
```

Comment:
It seems like btTransform& transform is an immutable variable. It's better to make it const.